src/resmom/stage_func.c

```diff
@@ -167,18 +167,24 @@ is_child_path(char *dir, char *path)
 		strncpy(fullpath, path, sizeof(fullpath));
 	}
 
+	dir_real = malloc(sizeof(char) * (MAXPATHLEN + 1));
+	fullpath_real = malloc(sizeof(char) * (2 * MAXPATHLEN + 2));
+
 	/* even if the file path is relative to some directory, */
 	/* it may use /../../ notation and escape that directory, */
 	/* so always perform full check of parent-child directory relation */
-	dir_real = realpath(dir, NULL);
-	fullpath_real = realpath(fullpath, NULL);
-
-	if (dir_real == NULL || fullpath_real == NULL)
-		goto error_exit;
 #ifdef WIN32
+	forward2back_slash(fullpath);
 	strncpy(dir_real, lpath2short(dir), sizeof(dir_real)-1);
 	strncpy(fullpath_real, lpath2short(fullpath), sizeof(fullpath_real)-1);
+#else
```

Comment:
The above Windows code looks incorrect because sizeof(dir_real) and sizeof(fullpath_real) will both return the size of the pointer, not the size of the malloc()'ed block.